controllers/openstackmachine_controller.go

```diff
@@ -348,6 +348,14 @@ func (r *OpenStackMachineReconciler) reconcileNormal(ctx context.Context, scope
 	}
 
 	addresses := instanceNS.Addresses()
+
+	// A node joins the cluster if the CSR generated by kubelet with the node name is approved.
```

Comment:
Can you make a note here that this is specific to OpenShift but likely also relevant to other systems.